src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -285,7 +274,19 @@ void CoreWorkerDirectTaskReceiver::HandlePushTask(
       task_main_io_service_.post([this]() { exit_handler_(); });
     } else {
       RAY_CHECK(objects_valid) << return_objects.size() << "  " << num_returns;
-      send_reply_callback(status, nullptr, nullptr);
+      if (task_spec.IsActorCreationTask()) {
+        rpc::WorkerAddress addr = {task_spec.GetMessage().caller_address().ip_address(),
+          task_spec.GetMessage().caller_address().port()};
+        auto client = std::shared_ptr<rpc::CoreWorkerClientInterface>(client_factory_(addr));
+
+        rpc::NotifyActorCreatedRequest request;
+        request.set_actor_creation_task_id(task_spec.TaskId().Binary());
+        request.mutable_address()->CopyFrom(rpc_address_);
+        RAY_CHECK_OK(client->NotifyActorCreated(request));
```

Comment:
When running release tests, I sometimes see this check fail with "failed to connect to all addresses". I'm not sure why this is happening, but it also seems error-prone to be doing an RPC within the task handler like this.

Instead of doing a separate RPC for when the actor is created, we should consider the actor created on successful reply back to the creator.

Also, we don't seem to ever reply to the RPC in this branch which is weird.

Reply:
thanks, I used rpc callback on reply instead.